src/main/java/org/mifos/pheevouchermanagementsystem/api/implementation/VoucherStatusApiController.java

```diff
@@ -0,0 +1,39 @@
+package org.mifos.pheevouchermanagementsystem.api.implementation;
+
+import java.util.concurrent.ExecutionException;
+import org.mifos.pheevouchermanagementsystem.api.definition.VoucherStatusApi;
+import org.mifos.pheevouchermanagementsystem.data.FetchVoucherResponseDTO;
+import org.mifos.pheevouchermanagementsystem.data.VoucherStatusResponseDTO;
+import org.mifos.pheevouchermanagementsystem.repository.VoucherRepository;
+import org.mifos.pheevouchermanagementsystem.service.FetchVoucherService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class VoucherStatusApiController implements VoucherStatusApi {
+
+    @Autowired
+    private FetchVoucherService fetchVoucherService;
+    @Autowired
+    private final VoucherRepository voucherRepository;
+
+    public VoucherStatusApiController(VoucherRepository voucherRepository) {
+        this.voucherRepository = voucherRepository;
+    }
+
+    @Override
+    public ResponseEntity<VoucherStatusResponseDTO> voucherStatus(String serialNumber, String registeringInstitutionId, String status)
+            throws ExecutionException, InterruptedException {
+        if (registeringInstitutionId == null || registeringInstitutionId.isEmpty() || !voucherRepository.existsBySerialNo(serialNumber)) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(null);
```

Comment:
the validation is not in proper format. Please make sure that the validation follows right structure